tests/vlan/test_vlan.py

```diff
@@ -31,44 +30,138 @@ def cfg_facts(duthosts, rand_one_dut_hostname):
+                if 'tagging_mode' not in vlan_members[k][port]:
+                    continue
+                mode = vlan_members[k][port]['tagging_mode']
+                config_ports_vlan[port].append({'vlanid':int(vlanid), 'ip':ip, 'tagging_mode':mode})
+
+    # For t0 topo, will add port to new VLAN, use 'orig' field to identify new VLAN.
+    vlan_id_list = [k for k, v in vlan_intfs_dict.items() if v['orig'] == False]
     pvid_cycle = itertools.cycle(vlan_id_list)
     # when running on t0 we can use the portchannel members
     if config_portchannels:
-        for po in config_portchannels.keys()[:2]:
-            vlan_ports_list.append({
+        portchannel_cnt = 0
+        for po in config_portchannels:
+            vlan_port = {
                 'dev' : po,
                 'port_index' : [config_port_indices[member] for member in config_portchannels[po]['members']],
-                'pvid' : pvid_cycle.next(),
-                'permit_vlanid' : [ vid for vid in vlan_id_list ]
-            })
+                'permit_vlanid' : []
+            }
+            if tbinfo['topo']['name'] == 't0-56-po2vlan' and po in config_ports_vlan:
+                vlan_port['pvid'] = 0
+                for vlan in config_ports_vlan[po]:
+                    if 'vlanid' not in vlan or 'ip' not in vlan or 'tagging_mode' not in vlan:
+                        continue
+                    if vlan['tagging_mode'] == 'untagged':
+                        vlan_port['pvid'] = vlan['vlanid']
+                    vlan_port['permit_vlanid'].append(vlan['vlanid'])
+            if tbinfo['topo']['name'] != 't0-56-po2vlan':
+                # Add 2 portchannels for test
+                if portchannel_cnt >= 2:
+                    continue
+                portchannel_cnt += 1
+                vlan_port['pvid'] = pvid_cycle.next()
+                vlan_port['permit_vlanid'] = vlan_id_list[:]
+            if 'pvid' in vlan_port:
+                vlan_ports_list.append(vlan_port)
 
     ports = [port for port in config_ports
         if config_port_indices[port] in ptf_ports_available_in_topo
         and config_ports[port].get('admin_status', 'down') == 'up'
         and port not in config_port_channel_member_ports]
 
-    for port in ports[:4]:
-        vlan_ports_list.append({
+    for i, port in enumerate(ports):
+        vlan_port = {
             'dev' : port,
             'port_index' : [config_port_indices[port]],
-            'pvid' : pvid_cycle.next(),
-            'permit_vlanid' : [ vid for vid in vlan_id_list ]
-        })
+            'permit_vlanid' : []
+        }
+        if tbinfo['topo']['name'] == 't0-56-po2vlan' and port in config_ports_vlan:
+            vlan_port['pvid'] = 0
+            for vlan in config_ports_vlan[port]:
+                if 'vlanid' not in vlan or 'ip' not in vlan or 'tagging_mode' not in vlan:
+                    continue
+                if '201911' not in duthost.os_version and vlan['tagging_mode'] == 'untagged':
```

Comment:
Why only test 201911 image? sonic-mgmt master branch will be used to test sonic-buildimage 201911 and later branches.